src/librustc/middle/ty/relate.rs

```diff
@@ -242,9 +336,12 @@ impl<'a,'tcx:'a> Relate<'a,'tcx> for ty::BareFnTy<'tcx> {
         let unsafety = try!(relation.relate(&a.unsafety, &b.unsafety));
         let abi = try!(relation.relate(&a.abi, &b.abi));
         let sig = try!(relation.relate(&a.sig, &b.sig));
-        Ok(ty::BareFnTy {unsafety: unsafety,
-                         abi: abi,
-                         sig: sig})
+        let RelateOk { value: ((unsafety, abi), sig), obligations } =
+            unsafety.chain(abi).chain(sig);
```

Comment:
Nothing wrong with this, but it might be mildly cleaner using the `into_value()` formulation I listed above. I guess which is cleaner will depend on your POV.